src/Avalonia.Controls/RepeatButton.cs

```diff
@@ -41,12 +42,14 @@ public int Delay
             set { SetValue(DelayProperty, value); }
         }
 
-        private void StartTimer()
+        private void StartTimer(RoutedEventArgs e)
         {
             if (_repeatTimer == null)
             {
                 _repeatTimer = new DispatcherTimer();
-                _repeatTimer.Tick += RepeatTimerOnTick;
+                _repeatTimer.Tick += (s, inner) => {
+                    RepeatTimerOnTick(s, e);
```

Comment:
Not sure we should be storing the event args for later like this. In particular this could be problematic for pointer events because the `Pointer` in the event args may not be valid outside the invocation of the particular pointer gesture IIRC, is that correct @kekekeks ?

I'd say that the first first click event (the one triggered in response to the actual pointer event) should have the `Inner` event set and subsequent ones should probably have a `null` inner event?

Reply:
That's tricky here as I wanted to mention the same.

Pointer will not be valid after user removed finger from the touch screen, that's true.
But in RepeatButton we have timer working only while button is still pressed. So, technically it's the same "gesture" with the same pointer. 
Similarly, with KeyDown event - same even will be pressed while timer is running.
Or at least in theory, as it should be double checked.

Reply:
Another tricky point. Think about a scenario when user hold shift key and press pointer. Then user released shift key but doesn't release pointer, then there will be no way to know from which click user has released shift key...

Reply:
@kekekeks what do you think?